### PR TITLE
[#1887] improvement(catalog-jdbc-common)(api): Change assertion to be a if-else check

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

change the assertion check to a if-else check.

### Why are the changes needed?

java assertion can be disabled in all packages and classes and it is not a safe op.
Reference java document: https://docs.oracle.com/en/java/javase/21/docs/specs/man/java.html#standard-options-for-java

Fix: [#1887](https://github.com/datastrato/gravitino/issues/1887)

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./gradlew clean :api:test :catalogs:catalog-jdbc-common:test`

### CODE REVIEW

**Comment by caican00** [comment]
cc @justinmclean

File: api/src/main/java/com/datastrato/gravitino/rel/TableChange.java [hunk 1]
```diff
@@ -581,8 +581,11 @@ final class After implements ColumnPosition {
     private final String column;
 
     private After(String column) {
-      assert column != null;
```
Comment (yuqi1129) [inline api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:584]: if (null == column) {
	throw new illlegalArumentExcepiotin("xxx");
}

this.column = column.

Comment (caican00) [reply api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:584]: got it, and i have updated it. Thank you for reviewing @yuqi1129

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by yuqi1129** [comment]
@caican00 
I noticed that there are some similarities left in some test classes like `TestByteUtils`, could you fix it by the way?

**Comment by caican00** [comment]
> @caican00 I noticed that there are some similarities left in some test classes like `TestByteUtils`, could you fix it by the way?

@yuqi1129 TestByteUtils is a test class, and I understand that we can keep an assertion in a test class. And what do you think about it?

**Comment by yuqi1129** [comment]
> > @caican00 I noticed that there are some similarities left in some test classes like `TestByteUtils`, could you fix it by the way?
> 
> @yuqi1129 TestByteUtils is a test class, and I understand that we can keep an assertion in a test class. And what do you think about it?

It's okay to use the word assert in test code, just think twice.

**Comment by jerryshao** [comment]
I think we should:

1. Avoid using assert in test, instead we should use Junit's assertion.
2. We'd better using Guava's `Preconditions`, rather than if..else..., which is not elegant.
3. We should check if javac or error-prone could check such code and throw an warning.

@caican00 @yuqi1129 can you do a follow-up PR to address what I mentioned here.

**Pull request merged**